docs/publishing/index.md

```diff
@@ -0,0 +1,125 @@
+
+## Do publish standard es modules
+
+In agreement with the "do" above, es modules are both Stage 4, supported in all modern browsers and tool-chains, ensure your package is published as es modules.
+
+## Do include `"main": "index.js"` and `"module": "index.js"` in your `package.json`
+
+Your `package.json` should have both its `main` entrypoint and `module` entrypoint point to the same es module file. You will find this already configured for you if you've started your project with `npm init @open-wc`, but ensure your `package.json` file includes something simialar to the following:
+
+```
+{
+  "main": "index.js",
+  "module": "index.js",
+  // ...
+}
+```
+
+## Do export element classes
+
+Every custom element is a class extension of `HTMLElement`; e.g. `class MyElement extends HTMLElement {}`. Be sure that you export this class from your package. When starting from `npm init @open-wc` you will see this export in `src/MyElement.js` and referenced by the `index.js` entry point. This will make it possible for both customization of your element via extension (e.g. `class NewElement extends MyElement {}`) as well as for advanced usage of the custom element in Scoped Registries, both [native](https://github.com/w3c/webcomponents/issues/716) (when available) or [synthetic](/scoped-elements/).
+
+## Do export side effects separately
+
+A [side effect](<https://en.wikipedia.org/wiki/Side_effect_(computer_science)>) in javascript is created when script changes state outside of the scope in which it is run. This includes when code in an es module alters values available on the `window` or `document`. A side effect occurs when registering Custom Elements via `customElements.define()`. JS files with side effects should exist separately from your packages entry point. `npm init @open-wc` takes care of this for you by creating a separate `my-element.js` that manages the registration of your custom element.
+
+## Do keep side effects out of your index.js
+
+When _not_ able to keep code with side effects out of your entry point, it should instead be put in a separate function that has to explicitly be called by the consumer (e.g. `run()`, `init()`, etc.). This prevents those side effects from occurring at run time.
```

Comment:
It doesnt really prevent the side effects from occurring at run time, they still occur whenever I opt in to calling `run()` myself, but at least as a user, I've made the conscious decision to call that function and register the side effects, and as author I've been explicit in my communication: "Hey, here's the side effectful stuff". 

I think the last sentence can be dropped